### ISSUE TITLE
False-positive `unreachable_code_linter` with `app$stop()` in a function

### ISSUE THREAD

**Issue by bersbersbers** [r-lib/lintr#1051]
```r
cleanup <- function() {
  app$stop()
  gc()
}
```

gives

> Code and comments coming after a top-level return() or stop() should be removed.

**Comment by AshesITR** [comment]
Thanks for the report.
I did a small technical investigation to aid with fixing:

The XML parse data for this MWE is

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<exprlist>
  <expr line1="1" col1="1" line2="4" col2="1" start="25" end="97">
    <expr line1="1" col1="1" line2="1" col2="7" start="25" end="31">
      <SYMBOL line1="1" col1="1" line2="1" col2="7" start="25" end="31">cleanup</SYMBOL>
    </expr>
    <LEFT_ASSIGN line1="1" col1="9" line2="1" col2="10" start="33" end="34">&lt;-</LEFT_ASSIGN>
    <expr line1="1" col1="12" line2="4" col2="1" start="36" end="97">
      <FUNCTION line1="1" col1="12" line2="1" col2="19" start="36" end="43">function</FUNCTION>
      <OP-LEFT-PAREN line1="1" col1="20" line2="1" col2="20" start="44" end="44">(</OP-LEFT-PAREN>
      <OP-RIGHT-PAREN line1="1" col1="21" line2="1" col2="21" start="45" end="45">)</OP-RIGHT-PAREN>
      <expr line1="1" col1="23" line2="4" col2="1" start="47" end="97">
        <OP-LEFT-BRACE line1="1" col1="23" line2="1" col2="23" start="47" end="47">{</OP-LEFT-BRACE>
        <expr line1="2" col1="3" line2="2" col2="12" start="51" end="60">
          <expr line1="2" col1="3" line2="2" col2="10" start="51" end="58">
            <expr line1="2" col1="3" line2="2" col2="5" start="51" end="53">
              <SYMBOL line1="2" col1="3" line2="2" col2="5" start="51" end="53">app</SYMBOL>
            </expr>
            <OP-DOLLAR line1="2" col1="6" line2="2" col2="6" start="54" end="54">$</OP-DOLLAR>
            <SYMBOL_FUNCTION_CALL line1="2" col1="7" line2="2" col2="10" start="55" end="58">stop</SYMBOL_FUNCTION_CALL>
          </expr>
          <OP-LEFT-PAREN line1="2" col1="11" line2="2" col2="11" start="59" end="59">(</OP-LEFT-PAREN>
          <OP-RIGHT-PAREN line1="2" col1="12" line2="2" col2="12" start="60" end="60">)</OP-RIGHT-PAREN>
        </expr>
        <expr line1="3" col1="3" line2="3" col2="6" start="75" end="78">
          <expr line1="3" col1="3" line2="3" col2="4" start="75" end="76">
            <SYMBOL_FUNCTION_CALL line1="3" col1="3" line2="3" col2="4" start="75" end="76">gc</SYMBOL_FUNCTION_CALL>
          </expr>
          <OP-LEFT-PAREN line1="3" col1="5" line2="3" col2="5" start="77" end="77">(</OP-LEFT-PAREN>
          <OP-RIGHT-PAREN line1="3" col1="6" line2="3" col2="6" start="78" end="78">)</OP-RIGHT-PAREN>
        </expr>
        <OP-RIGHT-BRACE line1="4" col1="1" line2="4" col2="1" start="97" end="97">}</OP-RIGHT-BRACE>
      </expr>
    </expr>
  </expr>
</exprlist>
```
So we need to exclude the  `SYMBOL_FUNCTION_CALL` if it has `preceding-sibling::OP-DOLLAR`.

**Comment by MichaelChirico** [comment]
awesome, thanks for early pre-release testing!
